src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -512,31 +478,16 @@ internal static PropertyValidationInfo GetPropertyValidationInfo(ModelBindingCon
                 return;
             }
 
-            if (value != null || TypeHelper.AllowsNullValue(property.PropertyType))
+            try
             {
-                try
-                {
-                    propertyMetadata.PropertySetter(bindingContext.Model, value);
-                }
-                catch (Exception exception)
-                {
-                    AddModelError(exception, bindingContext, dtoResult);
-                }
+                propertyMetadata.PropertySetter(bindingContext.Model, value);
             }
-            else
+            catch (Exception exception)
             {
-                // trying to set a non-nullable value type to null, need to make sure there's a message
-                var modelStateKey = dtoResult.Key;
-                var validationState = bindingContext.ModelState.GetFieldValidationState(modelStateKey);
-                if (validationState == ModelValidationState.Unvalidated)
-                {
-                    var errorMessage = Resources.ModelBinding_ValueRequired;
-                    bindingContext.ModelState.TryAddModelError(modelStateKey, errorMessage);
-                }
```

Comment:
Had some discussion - we think that this case means you've got a badly-behaving model binder. We're not going to special case it. I reduced the number of tests hitting this because it was overkill.